source/extensions/filters/http/ext_proc/ext_proc.h

```diff
@@ -330,8 +340,10 @@ class Filter : public Logger::Loggable<Logger::Id::ext_proc>,
   // know what response to return from certain failures.
   bool sent_immediate_response_ = false;
 
-  // Set to true then the mergePerRouteConfig() method has been called.
-  bool route_config_merged_ = false;
+  // Set a value when the mergePerRouteConfig() method has been called.
+  // This binds the constructed config to the life of the filter to extend the
+  // lifetimes of objects created during the merge.
+  absl::optional<FilterConfigPerRoute> route_config_merged_;
```

Comment:
This is the case for now, this is prelim work for the [dynamic metadata PR](https://github.com/envoyproxy/envoy/pull/30747) where I was having lifetime issues with the newly merged forwarding/receiving namespaces vectors, which are placed onto the processor state by const reference. We could copy the vectors in but that seems less than ideal since it's on the request path.

 I can punt this decision to that PR for more appropriately located discussion if desired.

Reply:
Yea, I feel it is better we defer this change to the PR that needs it.